fastjsonschema/generator.py

```diff
@@ -169,10 +181,30 @@ def generate_func_code(self, definition):
         Creates base code of validation function and calls helper
         for creating code by definition.
         """
-        with self.l('def func(data):'):
-            self.l('NoneType = type(None)')
+        self._validation_functions_done.add(self._resolver.get_uri())
+        self.l('NoneType = type(None)')
+        self.l('')
+        with self.l('def validate(data):'):
             self.generate_func_code_block(definition, 'data', 'data')
             self.l('return data')
+        # Generate parts that are referenced and not yet generated
+        while len(self._needed_validation_functions) > 0:
+            # Normal for ... in... loop not works here, because new functions
+            # can be found and therefore added in self._needed_validation_functions
+            # during it generation and dictionary cannot changed during iteration.
```

Comment:
Probably more clear would be:

During generation of validation function could be needed to generate new one added again to `_needed_validation_functions`. Therefore usage of while instead of for loop.